### DISCUSSION TITLE
Need some help with Deta Micros

### DISCUSSION THREAD

**Discussion by SlumberDemon** [deta/cloud-docs#274] [Help] [answered]
So, I am making an api using deta micros and fastapi, I added api keys and I wanted to know how to use the api key in the url in the web browser without using code. 

example: https://api.deta.dev/endpoint?key=APIKEY HERE
(this is how I tried it, but it doesn't work so how?)

sorry for dumb question.. x'D

**Comment by abdelhai** [comment]
hi, this doesn't work. it is also insecure to put any secrets in the url, esp api keys.
our intsructions on how to use micro api keys: https://docs.deta.sh/docs/micros/api_keys/#3-using-api-keys

**Reply by SlumberDemon** [reply]
I did see that, I want to access the fastapi /docs endpoint but I can only see it in a browser. When I have api keys enabled I can't access it. Is there another way?

**Comment by jnsougata** [comment, marked as answer]
Use `requesr.headers.get` to access the API KEY sent with request header ( example: `{'X-API-KEY': YOUR_API_KEY}` )
EXAMPLE:
```python
from fastapi import FastAPI, Request

@app.get("/")
async def root(request: Request):
    api_key = request.headers.get('X-API-KEY') 
   # do stuff with api_key here
```

**Reply by SlumberDemon** [reply]
thanks I'll try that
